internals/daemon/api_metrics.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"bytes"
+	"net/http"
+
+	"github.com/canonical/pebble/internals/logger"
+	"github.com/canonical/pebble/internals/metrics"
+	"github.com/canonical/pebble/internals/overlord/checkstate"
+	"github.com/canonical/pebble/internals/overlord/servstate"
+)
+
+func v1GetMetrics(c *Command, r *http.Request, _ *UserState) Response {
+	return metricsResponse{
```

Comment:
I realise I've done this elsewhere (like the files API), but you can actually simplify and do a handler like inline with:

```go
return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
    func v1GetMetrics(c *Command, r *http.Request, _ *UserState) Response {
        ...
    }
}
```